pkg/apis/crd/v1alpha1/types.go

```diff
@@ -291,3 +291,157 @@ type TLSProtocol struct {
 	// SNI (Server Name Indication) indicates the server domain name in the TLS/SSL hello message.
 	SNI string `json:"sni,omitempty"`
 }
+
+// Source describes the source spec of the traceflow.
+type Source struct {
+	// Namespace is the source namespace.
+	Namespace string `json:"namespace,omitempty"`
+	// Pod is the source pod.
+	Pod string `json:"pod,omitempty"`
+	// IP is the source IPv4 or IPv6 address. IP as the source is supported
+	// only for live-traffic Traceflow.
+	IP string `json:"ip,omitempty"`
+}
+
+// Destination describes the destination spec of the traceflow.
+type Destination struct {
+	// Namespace is the destination namespace.
+	Namespace string `json:"namespace,omitempty"`
+	// Pod is the destination pod, exclusive with destination service.
+	Pod string `json:"pod,omitempty"`
+	// Service is the destination service, exclusive with destination pod.
+	Service string `json:"service,omitempty"`
+	// IP is the destination IPv4 or IPv6 address.
+	IP string `json:"ip,omitempty"`
+}
+
+// IPHeader describes spec of an IPv4 header.
+type IPHeader struct {
+	// Protocol is the IP protocol.
+	Protocol int32 `json:"protocol,omitempty" yaml:"protocol,omitempty"`
```

Comment:
Do we support protocols out of TCP/UDP/ICMP? If not, do we require TransportHeader must be provided for TCP/UDP/ICMP and so this field is not really useful in that case? I am not suggesting any change, but trying to understand the current behaviors.

Reply:
only these 3 protocols and the source/dst port. We can support tcp flags eg but i'm not sure if this a popular choice.

Reply:
Could you answer this one too:
>  If not, do we require TransportHeader must be provided for TCP/UDP/ICMP and so this Protocol field is not really useful in that case?

Reply:
OK. i see your point. not sure if there will be any new comments on this (https://github.com/antrea-io/antrea/pull/5821#discussion_r1571239286). This did occured to me before. for now since `crdv1alpha1.Packet` is only used for `PacketCapture`, i prefer to remove IPHeader for now. Also TTL/flags is not supported for now and doesn't seeems like a must have future in the first phase.

Reply:
Sounds good to me to remove IPHeader in v1alpha1.